### PR TITLE
specs: turn all ivars into lets

### PR DESCRIPTION
**Pull request by chastell**
This is a follow-up to #638 – remove all instance variables in our specs and turn them into (lazy-loaded) `let` calls.

### CODE REVIEW

File: spec/reek/tree_walker_spec.rb [hunk 1]
```diff
@@ -5,9 +5,9 @@
 # Dummy repository to inject into TreeWalker in order to count statements in
 # all contexts.
 class TestSmellRepository
-  attr_reader :num_statements
+  attr_accessor :num_statements
   def examine(context)
-    @num_statements = context.num_statements
+    self.num_statements = context.num_statements
```
Comment (mvz) [inline spec/reek/tree_walker_spec.rb]: :smile:

**Comment by mvz** [comment]
Really nice to have this finally cleaned up! :+1:

**Comment by troessner** [comment]
@chastell you are truly a honeybadger on the spec refactoring battlefield.

File: spec/reek/tree_walker_spec.rb (hunk 1 above)
Comment (chastell) [inline spec/reek/tree_walker_spec.rb]: I was wondering whether anyone will review this PR this deeply – and I was not disappointed. :gift:

Comment (troessner) [inline spec/reek/tree_walker_spec.rb]: Yep, and I didn't do so because that statement made me realize that @mvz had already done that. :laughing:

**Pull request merged**